### PR TITLE
docs: Add section about unsupported go versions

### PR DESCRIPTION
**Pull request by AdamKorcz**
Fixes: https://github.com/google/oss-fuzz/issues/9022

@oliverchang PTAL

Signed-off-by: AdamKorcz <adam@adalogics.com>

### CODE REVIEW

File: docs/getting-started/new-project-guide/go_lang.md [hunk 1]
```diff
@@ -27,6 +27,14 @@ only. In that mode, fuzz targets for Go use the libFuzzer engine with native Go
 coverage instrumentation. Binaries compiled in this mode provide the same
 libFuzzer command line interface as non-Go fuzz targets.
 
+## Go version
+
+Go 1.16 and below is not supported. Projects with an unsupported Go version in their `go.mod` file will need to install a newer version in their OSS-Fuzz build scripts, for example by adding this line to their `Dockerfile`:
```
Comment (jonathanmetzman) [inline docs/getting-started/new-project-guide/go_lang.md:32]: I don't know anything about go but it's surprising to me that this is necessary.

**Review by jonathanmetzman — APPROVED** [review]
LGTM

File: docs/getting-started/new-project-guide/go_lang.md (hunk 1 above)
Comment (oliverchang) [reply docs/getting-started/new-project-guide/go_lang.md:32]: Also a little confusing to me. Why can't users just use the Go that we provide in our base image, which is up to date?

**Comment by AdamKorcz** [comment]
The linked issue was resolved here: https://github.com/google/oss-fuzz/pull/9067. I don't believe many - if any - other projects will have this issue. Closing.

**Pull request closed without merging**